nft_ingester/src/api/dapi/asset.rs

```diff
@@ -240,12 +240,16 @@ pub async fn get_by_ids<
         let mut urls_to_download = Vec::new();
 
         for (_, url) in asset_selected_maps.urls.iter() {
-            if urls_to_download.len() >= max_json_to_download {
-                break;
+            if url.is_empty() {
+                continue;
             }
-            if !asset_selected_maps.offchain_data.contains_key(url) && !url.is_empty() {
+            let offchain_data = asset_selected_maps.offchain_data.get(url);
+            if offchain_data.is_none() || offchain_data.unwrap().metadata.is_empty() {
```

Comment:
Seems not perfect to unwrap here. It's better to do `get(url)?.ok_or()` above. wdyt?